.github/workflows/data-plane-coverage.yaml

```diff
@@ -0,0 +1,43 @@
+name: Publish Rust Coverage to Codecov
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: data-plane
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+
+      - name: Install just and task
+        run: |
+          sudo snap install just --classic || true
+          curl -sSL https://taskfile.dev/install.sh | sh
+
+      - name: Install cargo-llvm-cov
+        run: cargo install cargo-llvm-cov
+
+      - name: Generate coverage report (lcov)
+        run: |
+          ../bin/task data-plane:test:coverage
+        working-directory: ./data-plane
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          files: ./data-plane/lcov.info
+          flags: rust
+          name: rust-data-plane
+          fail_ci_if_error: true
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
We could incorporate this step in the existing coverage job in https://github.com/agntcy/slim/blob/main/.github/workflows/reusable-rust-build-and-test.yaml#L176-L202

The reusable workflow needs to also be updated to accept the `CODECOV_TOKEN` as secret.

Reply:
Yup actually running it on PR is useful as well, I see it can also leave comments regarding how code coverage is affected: https://docs.codecov.com/docs/pull-request-comments

So we can leave the triggers as they are.

Reply:
check the same for agent directory by clicking on the README badge
https://github.com/agntcy/dir